coalib/settings/ConfigurationGathering.py

```diff
@@ -192,9 +193,11 @@ def load_configuration(arg_list, log_printer, arg_parser=None):
 
         sections = merge_section_dicts(sections, cli_sections)
 
-    for section in sections:
-        if section != 'default':
-            sections[section].defaults = sections['default']
+    for name, section in list(sections.items()):
+        section.set_default_section(sections)
+        if name == 'default':
+            logging.warning('Implicit \'Default\' section inheritance is '
+                            'deprecated. It will be removed soon.')
```

Comment:
I think there's no way the user can avoid this warning because ATM coala *always* creates a default section. I think we should change that.